core/src/main/java/org/elasticsearch/http/netty/NettyHttpRequest.java

```diff
@@ -69,6 +69,14 @@ public NettyHttpRequest(org.jboss.netty.handler.codec.http.HttpRequest request,
     @Override
     public Method method() {
         HttpMethod httpMethod = request.getMethod();
+
+        if (request.headers() != null) {
+            String methodOverride = request.headers().get("X-HTTP-Method-Override");
+            if (methodOverride != null) {
+                httpMethod = HttpMethod.valueOf(methodOverride);
+            }
+        }
+
```

Comment:
I agree. I think IMHO that it would make sense but in another PR. I mean this test thing should not shade that we want to support `X-HTTP-Method-Override` (Google) or `X-HTTP-Method` (Microsoft) or `X-METHOD-OVERRIDE` (IBM).

Going to look if we can do something else here in tests. Most likely we can in Java as httpclient solved it already :)